### PR TITLE
Add retry support 

### PR DESCRIPTION
**Pull request by HantingZhang2**
### What does this PR do?

Adds option to enable retry so that when a response code 429 is received the api client would retry 

### Review checklist

Please check relevant items below:

- [ ] This PR includes all [newly recorded cassettes](/DEVELOPMENT.md) for any modified tests.

- [ ] This PR does not rely on API client schema changes.
  - [ ] The CI should be fully passing.
- [ ] Or, this PR relies on API schema changes and this is a Draft PR to include tests for that new functionality.
  - Note: CI shouldn't be run on this Draft PR, as its expected to fail without the corresponding schema changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -165,6 +165,29 @@ api_instance.list_incidents_with_pagination() do |incident|
 end
 ```
 
+### Retry
+
+To enable the client to retry when rate limited (status 429) or status 500 and above:
+
+```ruby
+config = DatadogAPIClient::Configuration.new
+config.retry_config['enableRetry'] = true
```
Comment (therve) [inline README.md:174]: Need to update those references.

**Review by therve — CHANGES_REQUESTED** [review]

File: lib/datadog_api_client/configuration.rb [hunk 2]
```diff
@@ -230,6 +244,13 @@ def self.default
       @@default ||= Configuration.new
     end
 
+    def backoff_base=(value)
+      if value < 2
+        raise ArgumentError, 'backoff_base cannot be smaller than 2'
```
Comment (therve) [inline lib/datadog_api_client/configuration.rb:249]: May need a native speaker, but I would say `less than`

**Review by therve — APPROVED** [review]
Looks great!

File: lib/datadog_api_client/configuration.rb (hunk 2 above)
Comment (HantingZhang2) [reply lib/datadog_api_client/configuration.rb:249]: I think smaller than and less then are synonyms. Both works

**Pull request merged**